backend/pkg/serde/cbor.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package serde
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/fxamacker/cbor/v2"
+	cborsrv "github.com/redpanda-data/console/backend/pkg/cbor"
+	"strings"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+var cborDecoder = cbor.DecOptions{}
+
+var _ Serde = (*CborSerde)(nil)
+
+// CborSerde represents the serde for dealing with CBOR types.
+type CborSerde struct {
+	CborService *cborsrv.Service
+}
+
+// Name returns the name of the serde payload encoding.
+func (CborSerde) Name() PayloadEncoding {
+	return PayloadEncodingCbor
+}
+
+// DeserializePayload deserializes the kafka record to our internal record payload representation.
+func (d CborSerde) DeserializePayload(_ context.Context, record *kgo.Record, payloadType PayloadType) (*RecordPayload, error) {
+	if d.CborService == nil {
```

Comment:
If the service is removed we should check the config instead whether CBOR is enabled and the topic is enabled to test for deserialization

Reply:
Ok

Reply:
Replaced CborSvc with direct config usage